src/client/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -161,6 +161,11 @@ public partial class AuthenticationResult
             CorrelationId = correlationID;
             ApiEvent = apiEvent;
             AuthenticationResultMetadata = new AuthenticationResultMetadata(tokenSource);
+
+            if (msalAccessTokenCacheItem.RefreshOn.HasValue)
+            {
+                AuthenticationResultMetadata.RemainingTimeBeforeRefresh = msalAccessTokenCacheItem.RefreshOn.Value.Millisecond - DateTimeOffset.UtcNow.Millisecond;
```

Comment:
Please expose `DateTimeOffset? RefreshOn` to be consistent with AuthenticationResult.ExpiresOn